actionpack/lib/action_controller/metal/helpers.rb

```diff
@@ -112,6 +111,10 @@ def all_helpers_from_path(path)
         def all_application_helpers
           all_helpers_from_path(helpers_path)
         end
+
+        def renderer
```

Comment:
I think instead of defining a method here we could check if `renderer` is defined, if it is we call `renderer.helpers`, if not we fallback to the previous implementation. `renderer` would be implemented by https://github.com/rails/rails/blob/c2fcc6ac18f102d7650a416ec87bd65db6078a9d/actionpack/lib/action_controller/metal/rendering.rb#L15

Reply:
I'm trying to see in what case would `renderer` not be defined? 
`ActionController::Base` calls `setup_render!` [here](https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/base.rb#L256) which ends up defining a `renderer`: https://github.com/rails/rails/blob/82d184839e573b2085d3ea8007f81ab31827edd0/actionpack/lib/action_controller/metal/rendering.rb#L17-L19

Wont calling `ApplicationController.helpers` always load up `ActiionController::Base` and trigger the `setup_render!`?

Reply:
The case where that is not true is a controller that inherits from `ActionController::Metal` and includes `ActionController::Helpers` but not `Rendering`.

Reply:
ah yes, I see now, I'll amend the PR 👍